rpcserver.go

```diff
@@ -4524,16 +4524,55 @@ func (r *rpcServer) FeeReport(ctx context.Context,
 // 0.000001, or 0.0001%.
 const minFeeRate = 1e-6
 
+// checkMaxHtlc errors if the user tries to update their channel policy with
+// a max HTLC value that is too high or too low for a given list of channels.
+func checkMaxHtlc(maxHtlc uint64, channels []*channeldb.OpenChannel) error {
+	maxHtlcRequested := lnwire.MilliSatoshi(maxHtlc)
+	for _, ch := range channels {
+		if maxHtlcRequested < ch.LocalChanCfg.MinHTLC {
+			return fmt.Errorf("max htlc size of %v msat is below "+
+				"min HTLC policy of %v msat for channel %v",
+				maxHtlc, ch.LocalChanCfg.MinHTLC,
+				ch.FundingOutpoint)
+		}
+
+		// The max HTLC policy field must be less than or equal to the
+		// channel capacity AND less than or equal to the max in-flight
+		// HTLC value. Since the latter is always less than or equal to
+		// the former, only check the max in-flight value.
+		maxPendingMsats := ch.LocalChanCfg.ChannelConstraints.MaxPendingAmount
+		if maxPendingMsats < maxHtlcRequested {
+			return fmt.Errorf("max htlc size of %v msat is above "+
+				"max pending amount of %v msat for channel %v",
+				maxHtlc, maxPendingMsats, ch.ShortChannelID)
+		}
+	}
+
+	return nil
+}
+
 // UpdateChannelPolicy allows the caller to update the channel forwarding policy
 // for all channels globally, or a particular channel.
 func (r *rpcServer) UpdateChannelPolicy(ctx context.Context,
 	req *lnrpc.PolicyUpdateRequest) (*lnrpc.PolicyUpdateResponse, error) {
 
 	var targetChans []wire.OutPoint
+	var targetDbChans []*channeldb.OpenChannel
 	switch scope := req.Scope.(type) {
-	// If the request is targeting all active channels, then we don't need
-	// target any channels by their channel point.
+	// If the request is targeting all active channels, then we need to retrieve
+	// them to check that the request's max HTLC value is within range.
 	case *lnrpc.PolicyUpdateRequest_Global:
+		openChannels, err := r.server.chanDB.FetchAllOpenChannels()
+		if err != nil {
+			return nil, err
+		}
+
+		pendingChannels, err := r.server.chanDB.FetchPendingChannels()
```

Comment:
There is a small gap here if a channel moves from pending to open in between these calls.

Reply:
Added TODO :grimacing: